packages/consumption/src/modules/requests/itemProcessors/transferFileOwnership/TransferFileOwnershipRequestItemProcessor.ts

```diff
@@ -94,25 +116,30 @@ export class TransferFileOwnershipRequestItemProcessor extends GenericRequestIte
         requestInfo: LocalRequestInfo
     ): Promise<TransferFileOwnershipAcceptResponseItem> {
         const peerFile = await this.accountController.files.getOrLoadFileByReference(requestItem.fileReference);
-        const fileContent = await this.accountController.files.downloadFileContent(peerFile);
-
-        const ownFile = await this.accountController.files.sendFile({
-            buffer: fileContent,
-            title: peerFile.cache!.title,
-            description: peerFile.cache!.description,
-            filename: peerFile.cache!.filename,
-            mimetype: peerFile.cache!.mimetype,
-            expiresAt: CoreDate.from("9999-12-31T00:00:00.000Z"),
-            tags: peerFile.cache!.tags
-        });
+
+        let ownFile: File;
+        if (requestItem.ownershipToken) {
+            ownFile = await this.accountController.files.claimFileOwnership(peerFile.id, requestItem.ownershipToken);
+        } else {
+            const fileContent = await this.accountController.files.downloadFileContent(peerFile);
+            ownFile = await this.accountController.files.sendFile({
+                buffer: fileContent,
+                title: peerFile.cache!.title,
+                description: peerFile.cache!.description,
+                filename: peerFile.cache!.filename,
+                mimetype: peerFile.cache!.mimetype,
+                expiresAt: CoreDate.from("9999-12-31T00:00:00.000Z"),
+                tags: peerFile.cache!.tags
+            });
+        }
```

Comment:
we should stop creating this kind of code. there are many effective ways to avoid those floating non-const variables. In that case I'd heavily vote for extracting everything below to a private function and then you can also get rid of the else :)

Reply:
the more I think about it - you should extract this if else to a private function